test/unit/helpers/httpApi.js

```diff
@@ -0,0 +1,488 @@
+var chai = require('chai');
+var expect = require('chai').expect;
+var sinon = require('sinon');
+var _ = require('underscore');
+
+var httpApi = require('../../../helpers/httpApi');
+
+describe('httpApi', function () {
+	describe('middleware', function () {
+		describe('cors', function () {
+			it('headers are set correctly', function () {
+				var headers = {};
+				var called = false;
+				var res = {
+					header: function (key, value) {
+						headers[key] = value;
+					}
+				};
+				httpApi.middleware.cors(null, res, function () {called = true;});
+				expect(headers).to.eql({'Access-Control-Allow-Origin':'*','Access-Control-Allow-Headers': 'Origin, X-Objected-With, Content-Type, Accept'});
+				expect(called).to.be.true;
+			});
+		});
+
+		describe('errorLogger', function () {
+			var traceCalled = false;
+			beforeEach(function () {
+				sinon.stub(console,'trace').callsFake(function (err) {
+					traceCalled = true;
+				});
+			});
+			afterEach(function () {
+				console.trace.restore();
+				traceCalled = false;
+			});
+
+			it('error = null', function () {
+				var nextCalled = false;
+				httpApi.middleware.errorLogger(null, null, null, null, function () {nextCalled = true;});
+				expect(nextCalled).to.be.true;
+				expect(traceCalled).to.be.false;
+			});
+
+			it('error != null', function () {
+				var nextCalled = false;
+				var errorCalled = false;
+				var sendCalled = false;
+				var logger = {
+					error: function (message, err) {
+						expect(message).to.equal('API error api/url');
+						errorCalled = true;
+					}
+				};
+				var res = {
+					status: function (httpCode) {
+						expect(httpCode).to.equal(500);
+						return {
+							send: function (val) {
+								expect(val.success).to.be.false;
+								expect(val.error).to.equal('API error: forcedError');
+								sendCalled = true;
+							}
+						};
+					}
+				};
+
+				httpApi.middleware.errorLogger(logger, {message: 'forcedError'}, {url: 'api/url'}, res, function () {nextCalled = true;});
+				expect(nextCalled).to.be.false;
+				expect(errorCalled).to.be.true;
+				expect(sendCalled).to.be.true;
+				expect(traceCalled).to.be.true;
+			});
+		});
+
+		describe('logClientConnections', function () {
+			it('logs the message on the logger', function () {
+				var loggerCalled = false;
+				var nextCalled = false;
+				var logger = {
+					log: function (message) {
+						expect(message).to.equal('GET req/url from 127.0.0.1');
+						loggerCalled = true;
+					}
+				};
+				var req = {method: 'GET', url: 'req/url', ip: '127.0.0.1'};
+				httpApi.middleware.logClientConnections(logger, req, null, function () {nextCalled = true;});
+				expect(loggerCalled).to.be.true;
+				expect(nextCalled).to.be.true;
+			});
+		});
+
+		describe('blockchainReady', function () {
```

Comment:
check if isLoaded was called and adjust the test to returned results in seperate describe sections